### PR TITLE
Added development env setup instructions in documentation

### PR DESCRIPTION
**Pull request by smitthakkar96**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -263,6 +263,18 @@ The default GitHub Issues and Pull Requests interface.
...
 - Timeshift feature to compare 2 different sa binaries
 
+### Setting up development environment
+
+- Clone the repo
+- run cp ``` cp env.example .env ```
+- Start the containers with ``` docker compose up ```
+- Kill the web container ``` docker kill sarjitsu_web_1 ```
+- Navigate to ``` sarjitsu/lib/backend/src ```
+- Install dependencies with ``` sudo dnf install python3-devel && pip install -r requirements.txt ```
```
Comment (arcolife) [inline README.md]: fyi, I installed these too:
```
gpgme-devel
gpgme
```

File: README.md [hunk 2]
```diff
@@ -263,6 +263,18 @@ The default GitHub Issues and Pull Requests interface.
...
+- Modify CFG_PATH in config.py to ``` lib/backend/conf/sarjustu.ini.example ```
+- Set DEBUG=True in config.py
```
Comment (arcolife) [inline README.md]: and the manual changes we did to URLs for frontend, api etc.. linked as container names in config/init.py ?

File: README.md [hunk 3]
```diff
@@ -263,6 +263,18 @@ The default GitHub Issues and Pull Requests interface.
...
 
+### Setting up development environment
+
+- Clone the repo
+- run cp ``` cp env.example .env ```
+- Start the containers with ``` docker compose up ```
+- Kill the web container ``` docker kill sarjitsu_web_1 ```
+- Navigate to ``` sarjitsu/lib/backend/src ```
+- Install dependencies with ``` sudo dnf install python3-devel && pip install -r requirements.txt ```
+- Modify CFG_PATH in config.py to ``` lib/backend/conf/sarjustu.ini.example ```
```
Comment (arcolife) [inline README.md]: wouldn't that relative path depend on where we execute `python server.py` from ? more like an indication to user to set this per their env?

File: README.md (hunk 1 above)
Comment (arcolife) [inline README.md]: and activate virtual env :-)

sorry about nitpicking!

**Review by arcolife — CHANGES_REQUESTED** [review]

**Comment by arcolife** [comment]
@smitthakkar96 just added a few notes. Thanks for submitting the PR!

**Comment by arcolife** [comment]
ref from my env: https://gist.github.com/arcolife/8cceeca6f0a051a6312442a35fd5ad70#gistcomment-2171018

..in addition to https://github.com/distributed-system-analysis/sarjitsu/pull/35#discussion_r131864872

@smitthakkar96 side note, I was unable to convert the sa binary after these changes. so __perhaps__ an additional step needed about paths related to SA binary processing scripts in `backend/src/scripts` ?

File: README.md (hunk 1 above)
Comment (smitthakkar96) [reply README.md]: When I did on my fedora machine, I had to run dnf install python3-devel python3-gpgme and it worked.

**Comment by geetikabatra** [comment]
redis linkage in web service's config `__init__.py` could be avoided and instead, a small hack to /etc/hosts can be done through 
```
echo "$(docker inspect $(docker ps | grep redis | awk '{print $NF}') | egrep '\"IPAddress\": \"172' | awk -F'"' '{print $(NF-1)}') redis" | sudo tee -a /etc/hosts
```

**Comment by arcolife** [comment]
reworked into https://github.com/distributed-system-analysis/sarjitsu/pull/41

**Pull request closed without merging**